cmd/dlv/dlv_test.go

```diff
@@ -105,3 +108,60 @@ func TestBuild(t *testing.T) {
 		t.Fatal("Program did not exit")
 	}
 }
+
+func testIssue398(t *testing.T, dlvbin string, cmds []string) (stdout, stderr []byte) {
+	var stdoutBuf, stderrBuf bytes.Buffer
+	buildtestdir := filepath.Join(protest.FindFixturesDir(), "buildtest")
+	cmd := exec.Command(dlvbin, "debug")
+	cmd.Dir = buildtestdir
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		t.Fatal(err)
+	}
+	cmd.Stdout = &stdoutBuf
+	cmd.Stderr = &stderrBuf
+	if err := cmd.Start(); err != nil {
+		t.Fatal(err)
+	}
+	for _, c := range cmds {
+		fmt.Fprintf(stdin, "%s\n", c)
+	}
+	// ignore "dlv debug" command error, it returns
+	// errors even after successful debug session.
+	cmd.Wait()
+	stdout, stderr = stdoutBuf.Bytes(), stderrBuf.Bytes()
+
+	debugbin := filepath.Join(buildtestdir, "debug")
+	_, err = os.Stat(debugbin)
+	if err == nil {
+		t.Errorf("running %q: file %v was not deleted\nstdout is %q, stderr is %q", cmds, debugbin, stdout, stderr)
+		return
+	}
+	if !os.IsNotExist(err) {
+		t.Errorf("running %q: %v\nstdout is %q, stderr is %q", cmds, err, stdout, stderr)
+		return
+	}
+	return
+}
+
+func TestIssue398(t *testing.T) {
```

Comment:
For tests named after issues I like to have a small comment inside the test describing what the test is testing, so that we don't have to track down the issue to know why it was introduced. Do you mind adding something like "tests that the debug executable is removed after exit" here?